an-rope/src/internals.rs

```diff
@@ -55,20 +55,38 @@ impl BranchNode {
     fn split(self, index: usize) -> (Node, Node) {
         let weight = (&self).weight;
         if index < weight {
-            // split the left node
+            // if the index to split on is less than this node's weight,
+            // we split this node's left child.
```

Comment:
This is doing the thing again where the comment describes what is going on instead of what the intent is.